File: 10. Subprogram/B. Tukang Sulap/b_tukang_sulap.java
import java.util.Scanner;

class array {
    public int[][] arr;
    public array() {
        arr = new int[2][1001];
    }
}

class b_tukang_sulap {
    public static void swap(array swapArr, int P, int x, int Q, int y) {
        int temp = swapArr.arr[P][x];
        swapArr.arr[P][x] = swapArr.arr[Q][y];
        swapArr.arr[Q][y] = temp;
    }
    public static void main(String[] args) {
        array swapArr = new array();
        Scanner scan = new Scanner(System.in);
        int N = scan.nextInt();
        for (int i = 0; i < 2; i++) {
            for (int j = 0; j < N; j++) {
                int num = scan.nextInt();
                swapArr.arr[i][j] = num;
            }
        }
        int T = scan.nextInt();
        for (int i = 0; i < T; i++) {
            int x, y, p1, q1;
            char P, Q;
            P = scan.next().charAt(0);
            x = scan.nextInt();
            Q = scan.next().charAt(0);
            y = scan.nextInt();
            p1 = P - 'A';
            q1 = Q - 'A';
            swap(swapArr, p1, x - 1, q1, y - 1);
        }
        scan.close();
        for (int i = 0; i < 2; i++) {
            for (int j = 0; j < N; j++) {
                System.out.print(swapArr.arr[i][j] + " ");
            }
            System.out.println();
        }
    }
}

File: 10. Subprogram/B. Tukang Sulap/b_tukang_sulap.go
package main

import (
  "fmt"
)

func swap(a *int, b *int) {
  *a, *b = *b, *a
}

func main() {
  var N, T, x, y, p1, q1 int
  var arr [2][1001]int
  var P, Q string
  fmt.Scanln(&N)
  for i := 0; i < 2; i++ {
    for j := 0; j < N; j++ {
      if j < N-1 {
        fmt.Scan(&arr[i][j])
      } else {
        fmt.Scanln(&arr[i][j])
      }
    }
  }
  fmt.Scanln(&T)
  for _ = 0; T > 0; T-- {
    fmt.Scan(&P, &x, &Q, &y)
    x -= 1
    y -= 1
    p1 = int(P[0] - 'A')
    q1 = int(Q[0] - 'A')
    swap(&arr[p1][x], &arr[q1][y])
  }
  for i := 0; i < 2; i++ {
    for j := 0; j < N; j++ {
      fmt.Print(arr[i][j], " ")
    }
    fmt.Println()
  }
}
